src/pipecat/services/background_noise.py

```diff
@@ -0,0 +1,120 @@
+import asyncio
+import json
+import time
+from asyncio import sleep
+from io import BytesIO
+
+import loguru
+
+from pipecat.processors.frame_processor import FrameProcessor, FrameDirection
+from pydub import AudioSegment
+from pipecat.frames.frames import AudioRawFrame, OutputAudioRawFrame, Frame, BotStartedSpeakingFrame, \
+    BotStoppedSpeakingFrame, EndFrame
+
+
+class BackgroundNoiseEffect(FrameProcessor):
+    def __init__(self, websocket_client, stream_sid, music_path):
+        super().__init__(sync=False)
+        self._speaking = True
+        self._audio_task = self.get_event_loop().create_task(self._audio_task_handler())
+        self._audio_queue = asyncio.Queue()
+        self._stop = False
+        self.stream_sid = stream_sid
+        self.websocket_client = websocket_client
+        self.music_path = music_path
+        self.get_music_part_gen = self._get_music_part()
+        self.emptied = False
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, BotStartedSpeakingFrame):
+            self._speaking = True
+
+        if isinstance(frame, BotStoppedSpeakingFrame):
+            self._speaking = False
+            self.emptied = False
+
+        if isinstance(frame, AudioRawFrame) and self._speaking:
+            if not self.emptied:
+                self.emptied = True
+                buffer_clear_message = {"event": "clear", "streamSid": self.stream_sid}
+                await self.websocket_client.send_text(json.dumps(buffer_clear_message))
+
+            frame.audio = self._combine_with_music(frame)
+
+        if isinstance(frame, EndFrame):
+            self._stop = True
+
+        await self.push_frame(frame, direction)
+
+    def _combine_with_music(self, frame: AudioRawFrame):
+        """
+        Combines small raw audio segments from the frame with chunks of a music file.
+        """
+        small_audio_bytes = frame.audio
+        music_audio = AudioSegment.from_wav(self.music_path)
```

Comment:
we don't need to load the audio segment here since we could now pass it in the construtor. This way we can use `from_mp3` for example.